ext/js/language/translator.js

```diff
@@ -415,51 +414,44 @@ export class Translator {
         }
     }
 
-    // Deinflections and text transformations
+    // Deinflections and text preprocessing
 
     /**
      * @param {string} text
      * @param {import('translation').FindTermsOptions} options
      * @returns {import('translation-internal').DatabaseDeinflection[]}
      */
     _getAlgorithmDeinflections(text, options) {
-        /** @type {import('translation-internal').TextDeinflectionOptionsArrays} */
-        const textOptionVariantArray = [
-            this._getTextReplacementsVariants(options),
-            this._getTextOptionEntryVariants(options.convertHalfWidthCharacters),
-            this._getTextOptionEntryVariants(options.convertNumericCharacters),
-            this._getTextOptionEntryVariants(options.convertAlphabeticCharacters),
-            this._getTextOptionEntryVariants(options.convertHiraganaToKatakana),
-            this._getTextOptionEntryVariants(options.convertKatakanaToHiragana),
-            this._getCollapseEmphaticOptions(options)
-        ];
+        const textPreprocessors = this._getTextPreprocessors(options);
```

Comment:
Can `textPreprocessors` be constructed in prepare and cached so it doesn't need to be done every time? Same for the other two maps below.

Reply:
I've moved it to prepare, I guess it's fine to do it for all languages since it's only done once and it's probably pretty fast. 

I left the third `variantSpace` there, because I think the `textReplacements` might change after prepare?

Reply:
Good point